### PR TITLE
Issue326 Redo Add Join Staff Member form

### LINKED ISSUE #326: Redo Add Join Staff Member form
**Purpose**
A formik form to actually create a new staff member (right now it's sending dummy data)

**Proposed Solution**
Use a formik form to collect the necessary info and send it to the backend.
Should be done after issue #322 

**Acceptance Criteria**
Please list the requirements the implementation of this feature should meet.
- [ ] Add a formik form to match the mockups
- [ ] Add validation to fields
- [ ] Send the info to the /user/invite endpoint on the backend (requires #322 )

**Mockup Screenshot or Page Number**
Paste a screenshot of the feature from the FIGMA mockups or cite the page number.

**Additional context**

### PR DESCRIPTION
**Pull request by Chris-Boe**
### What issue is this solving?
Please connect this Pull Request to the issue it is resolving by using the controls on the right side of this PR.

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? n
Any special requirements to test? n

### Please make sure you've attempted to meet the following coding standards
- [x] Code builds successfully
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code
- [x] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

**Comment by Chris-Boe** [comment]
Users submitted through this form get added to the Pending list on the dashboard, which is where you assign a role to the user. So I don't think we actually need the Make Admin checkbox like the mockup says.
@htharker42 thoughts?

File: src/views/joinStaff/joinStaff.jsx [hunk 1]
```diff
@@ -15,32 +15,39 @@ export const JoinStaff = () => {
...
-    );
+    return user ?
+      (
+        <JoinStaffCard
+          key={user.id}
+          name={`${user.firstName} ${user.lastName}`}
+          phoneNumber={user.phone}
+          email={user.email}
+          tickets={user.tickets}
+          tenants={user.tenants}
```
Comment (NickSchimek) [inline src/views/joinStaff/joinStaff.jsx]: I'm getting console errors when viewing the `/staff` endpoint. 
```
Invalid prop `tenants` of type `string` supplied to `JoinStaffCard`, expected `number`.
    in JoinStaffCard (at joinStaff.jsx:41)
```

**Review by NickSchimek — DISMISSED** [review]
Looks good. Two comments about viewing the index page. There are console errors as mentioned below, and the `reassign` (broken link) formatting needs is off too.
Screen Shot 2021-05-31 at 8 55 02 AM

Although those two issues are probably not related to this ticket.

One question related to this issue. What should happen when the user clicks save? Should they be taken back to the index `/staff` page? Because as it is now where it doesn't redirect feels kind of weird. Your left with an empty form that has the options to save/cancel after a successful event. I guess this is something the end user can expand upon too once we get this out to them.

File: src/views/joinStaff/joinStaff.jsx (hunk 1 above)
Comment (Chris-Boe) [reply src/views/joinStaff/joinStaff.jsx]: I'll fix this, thanks for finding it!

**Comment by Chris-Boe** [comment]
> Looks good. Two comments about viewing the index page. There are console errors as mentioned below, and the `reassign` (broken link) formatting needs is off too.
> Screen Shot 2021-05-31 at 8 55 02 AM
> 
> Although those two issues are probably not related to this ticket.
> 
> One question related to this issue. What should happen when the user clicks save? Should they be taken back to the index `/staff` page? Because as it is now where it doesn't redirect feels kind of weird. Your left with an empty form that has the options to save/cancel after a successful event. I guess this is something the end user can expand upon too once we get this out to them.

I did think it felt a little weird, but we are not very consistent about it through the app so I'll ask to see what we want to do and we can make sure every page behaves the same!

File: src/views/joinStaff/joinStaff.jsx (hunk 1 above)
Comment (Chris-Boe) [reply src/views/joinStaff/joinStaff.jsx]: So looks like our backend is missing some info for the user, so I'll remove the props for this for now
image

**Review by htharker42 — APPROVED** [review]

**Pull request merged**